onnxruntime/python/tools/transformers/convert_generation.py

```diff
@@ -249,6 +249,15 @@
     )
     model_group.set_defaults(past_present_share_buffer=False)
 
+    model_group.add_argument(
+        "--use_decoder_masked_multihead_attention",
+        required=False,
+        action="store_true",
+        help="Uses `DecoderMaskedMultiheadAttention` to optimize the unidirectional decoding Attention computation. "
+        "Must be used with `past_present_share_buffer`",
```

Comment:
>"Must be used with `past_present_share_buffer`", [](http://example.com/codeflow?start=8&length=48)

check past_present_share_buffer in the following code before switching from Attention to DecoderMaskedMultiheadAttention

Reply:
This logic already exists - 

![image](https://user-images.githubusercontent.com/9969784/222538506-2f1e2a8e-ef87-40f8-b401-e6a0005a5466.png)